pkg/config/settings/runtime_setting_profiling.go

```diff
@@ -72,10 +74,7 @@ func (l ProfilingRuntimeSetting) Set(v interface{}) error {
 		// Note that we must derive a new profiling.Settings on every
 		// invocation, as many of these settings may have changed at runtime.
 		v, _ := version.Agent()
-		service := "datadog-agent"
```

Comment:
The logic for overriding the service name based on the flavor is now removed in favor of having clients pass in the service name explicitly